src/julia.h

```diff
@@ -2289,6 +2289,16 @@ typedef struct _jl_task_t {
     // uint8_t padding1;
     // multiqueue priority
     uint16_t priority;
+    // flag indicating whether or not this to measure the wall and cpu time of this task
+    uint8_t is_timing_enabled;
+    // timestamp this task first entered the run queue (TODO: int32 of ms instead?)
+    uint64_t first_enqueued_at;
```

Comment:
Do you think it's valuable to also have created_at?

It _might_ be interesting in some cases to see the difference between the created-at and first-scheduled-at timestamps? Maybe someone would be surprised to see that gap being large? Not sure though.

Reply:
yeah, maybe, but i didn't particularly want to add things i wasn't adding a public API for... what do you think? worth adding? if so, worth adding a public API?

Reply:
+1 that it should either have a public API or not exist at all.

🤔 My instinct was to say "just add it," but i know that we want to keep the struct smallish if possible for data-locality so I think you're right to leave it off unless we have a use for it.

Reply:
i've not added it in this PR, but seems trivial to add in a follow up if we decide it'd be worth having (we can add a field that gets set in `jl_new_task`)